builtin/plugins/dposv3/dpos.go

```diff
@@ -877,6 +884,44 @@ func (c *DPOS) RegisterCandidate(ctx contract.Context, req *RegisterCandidateReq
 	return c.emitCandidateRegistersEvent(ctx, candidateAddress.MarshalPB(), req.Fee)
 }
 
+func (c *DPOS) Unjail(ctx contract.Context, req *UnjailRequest) error {
+	if !ctx.FeatureEnabled(loomchain.DPOSVersion3_3, false) {
+		return errors.New("DPOS v3.3 is not enabled")
+	}
+	ctx.Logger().Info("DPOSv3 Unjail", "request", req)
+
+	candidateAddress := ctx.Message().Sender
+
+	// if req.Validator is not nil, make sure that the caller is the oracle
+	// only the oracle can unjail other validators, a validator can only unjail itself
+	if req.Validator != nil {
+		state, err := LoadState(ctx)
+		if err != nil {
+			return err
+		}
+		if state.Params.OracleAddress == nil || ctx.Message().Sender.Compare(loom.UnmarshalAddressPB(state.Params.OracleAddress)) != 0 {
+			return errors.New("Only the oracle can unjail other validators")
+		}
+		candidateAddress = loom.UnmarshalAddressPB(req.Validator)
+	}
+
+	statistic, err := GetStatistic(ctx, candidateAddress)
+	if err != nil || statistic == nil {
+		return errStatisticNotFound
+	}
+
+	if !statistic.Jailed {
+		return errors.New("Cannot unjail the validator that is not jailed")
```

Comment:
-> `fmt.Errorf("%s is not jailed", candidateAddress.String())

Reply:
Done